client/sysinfo/src/sysinfo.rs

```diff
@@ -32,14 +33,117 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Used to represent the unit used in the benchmarks.
+pub enum Unit {
+	GiBs,
+	MiBs,
+	KiBs,
+}
+
+impl fmt::Display for Unit {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.write_str(match self {
+			Unit::GiBs => "GiBs",
+			Unit::MiBs => "MiBs",
+			Unit::KiBs => "KiBs",
+		})
+	}
+}
+
+/// Throughput as measured in bytes per second.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Throughput(f64);
+
+const KIBIBYTE: f64 = 1024.0;
+
+impl Throughput {
+	/// `f64` kibibyte/s to byte/s.
+	pub fn from_kibs(kibs: f64) -> Throughput {
+		Throughput(kibs * KIBIBYTE)
+	}
+
+	/// `f64` mebibyte/s to byte/s.
+	pub fn from_mibs(mibs: f64) -> Throughput {
+		Throughput(mibs * KIBIBYTE * KIBIBYTE)
+	}
+
+	/// `f64` gibibyte/s to byte/s.
+	pub fn from_gibs(gibs: f64) -> Throughput {
+		Throughput(gibs * KIBIBYTE * KIBIBYTE * KIBIBYTE)
+	}
+
+	/// [`Self`] as f64.
+	pub fn as_f64(&self) -> f64 {
```

Comment:
```suggestion
	/// [`Self`] as number of byte/s.
	pub fn as_byte(&self) -> f64 {
```

Reply:
Considering the other function names maybe it would be better to call this `as_bytes`, or not?

Reply:
:+1: